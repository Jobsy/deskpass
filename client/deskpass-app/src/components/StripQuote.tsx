

import { useState } from "react";

import { onChange } from "../helperFunctions";
import { StripQuoteParagraph, StripQuoteTextarea, StripQuoteWrapper, TooltipTarget } from "../styled";
import ToolTip from "./ToolTip";

const StripQuote = () => {
  const [value, setValue] = useState("");
  const [isStripe, setIsStripe] = useState("");

  return (
    <StripQuoteWrapper data-testid="stripe-quote">
      <ToolTip 
        tooltipText="Enter a word or sentence containg quotes into the textbox, and get the quotes striped"
        position="bottom" 
        background="00ADB5"
        styleMe={false}
        >
        <TooltipTarget><h2>Word-Quote Striper</h2></TooltipTarget>
      </ToolTip>
      <StripQuoteTextarea
        value={value}
        autoComplete="on"
        placeholder="Enter quote"
        onChange={(event) => {
        onChange(event, setValue, setIsStripe);
        }}
      />
      <StripQuoteParagraph isStripe={!!isStripe}>
        {!!isStripe ? "Striped Quote:" : "Invalid!"} {isStripe}
      </StripQuoteParagraph>
    </StripQuoteWrapper>
  );
};

export default StripQuote;