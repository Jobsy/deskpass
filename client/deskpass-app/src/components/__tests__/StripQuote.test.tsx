import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer"
import StripQuote from "../StripQuote";


afterEach(() => {
    cleanup();
})

test('should render StripQuote component', () => {
    render(<StripQuote />);
    const stripQuoteComponent = screen.getByTestId("stripe-quote")
    
    expect(stripQuoteComponent).toBeInTheDocument();
});

test('should match snapshot', () => {
    const tree = renderer.create(<StripQuote />).toJSON()

    expect(tree).toMatchSnapshot()
});

export {}