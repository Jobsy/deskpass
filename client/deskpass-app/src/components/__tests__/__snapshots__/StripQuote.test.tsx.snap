// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="sc-bczRLJ eJKZJo"
  data-testid="stripe-quote"
>
  <div
    className="sc-hKMtZM IMoHK"
    data-testid="tool-tip"
  >
    <button
      className="sc-eCYdqJ jWFya"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <button
        className="sc-eCYdqJ jWFya"
      >
        <h2>
          Word-Quote Striper
        </h2>
      </button>
    </button>
  </div>
  <textarea
    autoComplete="on"
    className="sc-gsnTZi gKEenm"
    onChange={[Function]}
    placeholder="Enter quote"
    value=""
  />
  <p
    className="sc-dkzDqf ejhFWH"
  >
    Invalid!
     
  </p>
</div>
`;
